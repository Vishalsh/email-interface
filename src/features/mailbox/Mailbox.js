import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import { Route } from 'react-router-dom';

import mailboxActions from './mailbox.actions';
import loadingStates from 'constants/loadingStates';
import Emails from 'features/emails/Emails';
import NoEmail from 'features/emails/noEmail/NoEmail';
import EmailDetails from 'features/emails/emailDetails/EmailDetails';
import CreateEmail from 'features/emails/createEmail/CreateEmail';
import MailboxHeader from 'features/mailbox/mailboxHeader/MailboxHeader';
import classes from './Mailbox.module.scss';

export class Mailbox extends Component {
  componentDidMount() {
    const { match, getEmails, mailbox } = this.props;

    if (mailbox.loadingState === loadingStates.AT_REST) {
      getEmails(match.params.mailbox);
    }
  }

  componentWillReceiveProps(nextProps) {
    const { match, getEmails, mailbox } = nextProps;

    if ((match.params.mailbox !== this.props.match.params.mailbox) && mailbox.loadingState === loadingStates.AT_REST) {
      getEmails(match.params.mailbox);
    }
  }

  onClickDeleteEmails = () => {
    const { match, deleteEmails } = this.props;

    deleteEmails(match.params.mailbox);
  };

  render() {
    const { match, mailbox, isCreateEmailPopupOpen } = this.props;

    return (
      <section className={classes.mailbox}>
        <MailboxHeader mailbox={match.params.mailbox} onClickDeleteEmails={this.onClickDeleteEmails}/>

        {
          mailbox.emails.length > 0 ?
            <Emails mailbox={match.params.mailbox} ids={mailbox.emails}/>
            :
            <NoEmail />
        }

        <Route path={`${match.url}/:id`} component={EmailDetails}/>

        {
          isCreateEmailPopupOpen &&
          <CreateEmail />
        }
      </section>
    );
  }
}

Mailbox.defaultProps = {
  mailbox: {
    emails: [],
    loadingState: loadingStates.AT_REST
  },
  isCreateEmailPopupOpen: false
};

Mailbox.propTypes = {
  match: PropTypes.object.isRequired,
  mailbox: PropTypes.shape({
    emails: PropTypes.array,
    loadingState: PropTypes.string,
  }),
  isCreateEmailPopupOpen: PropTypes.bool.isRequired,
  getEmails: PropTypes.func.isRequired,
  deleteEmails: PropTypes.func.isRequired,
};

const mapStateToProps = (store, ownProps) => ({
  mailbox: store.mailbox[ownProps.match.params.mailbox],
  isCreateEmailPopupOpen: store.emails.isCreateEmailPopupOpen
});

const mapDispatchToProps = dispatch => bindActionCreators({
  getEmails: mailboxActions.getEmails,
  deleteEmails: mailboxActions.deleteEmails
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Mailbox);
