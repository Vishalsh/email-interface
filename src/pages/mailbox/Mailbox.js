import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';

import routes from 'constants/routes';
import SettingsCog from 'components/settingsCog/SettingsCog';
import MailboxAside from 'features/mailboxAside/MailboxAside';
import MailboxFeature from 'features/mailbox/Mailbox';

const Mailbox = (props) => {
  const { match } = props;

  return (
    <div className="row">
      <div className="col-sm-3">
        <MailboxAside />
      </div>
      <div className="col-sm-9">
        <Switch>
          <Redirect exact from={routes.MAILBOX} to={routes.INBOX}/>
          <Route path={`${match.url}/:mailbox`} component={MailboxFeature}/>
        </Switch>
      </div>
      <SettingsCog />
    </div>
  );
};

export default Mailbox;
