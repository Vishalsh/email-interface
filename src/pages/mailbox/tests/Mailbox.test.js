import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Mailbox from '../Mailbox';

describe('Mailbox', () => {
  const props = {
    match: {
      url: 'mailbox'
    }
  };

  it('should render the component', () => {
    expect(toJson(shallow(<Mailbox {...props}/>))).toMatchSnapshot();
  })
});