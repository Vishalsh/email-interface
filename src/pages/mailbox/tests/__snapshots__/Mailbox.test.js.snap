// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mailbox should render the component 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-3"
  >
    <Connect(MailboxAside) />
  </div>
  <div
    className="col-sm-9"
  >
    <Switch>
      <Redirect
        exact={true}
        from="/mailbox"
        push={false}
        to="/mailbox/inbox"
      />
      <Route
        component={[Function]}
        path="mailbox/:mailbox"
      />
    </Switch>
  </div>
</div>
`;
